src/composition/supergraph/binary.rs

```diff
@@ -95,6 +96,19 @@ impl SupergraphBinary {
         supergraph_config_path: Utf8PathBuf,
     ) -> Result<CompositionSuccess, CompositionError> {
         let args = self.prepare_compose_args(output_target, &supergraph_config_path);
+
+        if let OutputTarget::File(path) = output_target {
+            let parent = path.parent();
+            if let Some(parent) = parent {
+                if !parent.exists() {
+                    fs::create_dir_all(parent).map_err(|err| CompositionError::UpsertFile {
```

Comment:
> Recursively create a directory and all of its parent components if they are missing.

Reply:
Just a note. I looked this up to ensure the method did what I expected and pasted this clarification doc for any future readers.